### ISSUE TITLE
Infinite universe

### ISSUE THREAD

**Issue by agb32** [naev/naev#1204]
Not so much an issue, but to open discussion about an idea for the main Za'lek mission set.

Synopsis - Zalek scientists have developed a tangible simulated universe which they need exploring and reporting back on.  They are trying to understand "The Event" better, and also to create a perfect universe for their expanding populace.  In doing this, they have used vast resources, which if instead used on their space force could have led them crushing all the other factions.  But as they are scientists, they prefer peace and exploration.  In much the same way that scientists these days use the largest supercomputers to simulate the universe.  Except, due to their advanced tech, the simulated universe Za'lek have created can actually be entered.

This will be a procedurally generated infinite universe, and will allow for a huge amount of exploration.

When accessing it via jump point from Zalek space, the current planet stack is ignored, and the infinite one used instead.  The player is then in an alternative universe.

This will be somewhat simplified compared to the Naev universe:

Friendly faction (Za'lek, traders and miners), aggressive faction (the scientists aren't really sure how that got there).  
Upon entering, you start in a friendly area, and as get further from it, becomes more aggressive (up to a limit, beyond which it becomes less dangerous).
  - Heading north becomes more and more small enemy ships (rather like heading into collective territory).
  - Heading south gives larger and larger ships (eventually a few pirate kestrels per system).  All these ships are part of the
simulation of course - and it is explained that you might meet hostile ships, but that they're not sure how many.

Some planets can be landable.  These have no bar or missions, outfits or shipyards, though do have commodities, so trading within the simulated universe, or between real and simulated, can also be done.

Gas giants can be landed on to refuel.

Mission 1: Search for an exit jump point.  This will be located within 4 jumps of the start, and the player will be told this.

Mission 2: Map out N systems (e.g. 30) and report back.  If already mapped out this many during the first mission, then map out 10 more.  (numbers can be tweaked).

Mission 3: Head to a particular system (or area) to search out some feature that the scientists don't understand.  Once you arrive, you get some description.  Also see whether there are any hostilities.

Mission 4: Sample the extent of the simulated universe - player has to head to a certain sector, say 50 jumps North.

Mission 5: Similarly, 50 jumps South - all under the guise of scientific exploration.

Mission 6: Collect information about some artefacts spread over a number of systems.  some could be physical objects, others could be photographs/scientific information (i.e. just navigate to a particular point).  Before accepting the mission, need to check that you have enough cargo space to hold these items - so can't be done using a small fast ship.  Therefore some combat will be required even if this has previously been mostly avoided.  Upon completion, you are awarded the title "Chief Scientist (Exploration)".

Between each mission, the jump point will be removed - so that the player can only access the simulated universe when doing a mission.  However, upon completion of all missions, the jump point will remain open (accessible by the Chief Scientists), and the player is welcome to explore/trade/mine the infinite simulated universe.  This should appeal to players who enjoy exploration.  

We could also randomly leave outfits floating around for collection, though probably not.  Potentially, could also have another jump point from a pirate/collective system, which the Za'lek didn't know about or anticipate - which could explain the hostilities.

Mechanics:
Upon arrival in a system, all assets and jump points are discovered
(so no need to keep track of known assets).
Player savefile only needs to keep a record of systems visited,
probably the x,y coord of the system, so typically 8 bytes per system.
Additional code could be kept fairly small, I think.  Obviously can't all be done in Lua - and changes will need to be made to space.c, map.c, info.c etc - but I think this can be done in a fairly clean way.
Systems within map screen will be generated on the fly (no infinite system_stack), depending whether they're in view and whether they've been visited.

So - what do people think - does a mini-exploration sub-system within Naev sound like something that would add to the game?  Obviously, details of the missions themselves could be tweaked... added, deleted, story changed...

**Comment by onpon4** [comment]
Wouldn't that just be like a less good version of Nox Imperii? I've not played it but procedurally generated infinite universe I think is an apt description of what that's like.

🕵️

**Comment by nenau-again** [comment]
>Wouldn't that just be like a less good version of Nox Imperii?

Exactly, but Nox Imperii's automatically generated infinite universe is so cool that it's in my opinion worth to add a less good version of it to naev. The only concern for me are:

* To make it so that the mechanics is manageable in terms of performance and size of savefile
* To find somehow a way to link that to the main story with the hypergates and the proteron and such.

**Comment by agb32** [comment]
I guess it wouldn't necessarily need to be a less good version of Nox (looks nice - haven't played it) - we could add features as we go along, and perhaps introduce procedurally generated outfitters/shipyards from the start.  Could even have some of the more generic missions there.
Mechanics would be fine in terms of performance - no problem there.  For the save file, it just requires 8 bytes per system visited (as a binary, so typically 30% larger if encoded in Base64 for html).  But small anyway.

A link and hypergates into the real world at different points would also be possible, as part of the storyline.  Could even by the way by which Proteron make it into the main Naev world...

**Comment by onpon4** [comment]
Hmmmm, my concern is that a procedurally generated infinite universe would be a totally different kind of beast from what Naev is. Kind of like developing two games in parallel; Naev is already a huge game as it is, I don't know how well we could manage basically building in a variant of Nox Imperii too as a side thing.

Not to mention, making this a Za'lek thing might give the Za'lek too much importance. So if we were to do this then I would argue that it should be faction-neutral.

The best idea if we were to do this, I think, would be for the "alternate infinite universe" to be accessible via hypergates. Then it would be faction-neutral and depending on how hypergates work, it might be easier to implement that way. Even then, though, I'm still kinda iffy about the idea. It seems to me that a procedurally generated universe and hand-built universe are different enough that they should be two separate games.

🕵️

**Comment by Zireael07** [comment]
+1 to making it a separate mode accessible via hypergates and not tied to any faction/storyline.

**Comment by agb32** [comment]
Perhaps the Za'lek create it, but then after that its faction neutral with a few hypergates that they got the calculations wrong for!

**Comment by agb32** [comment]
Thinking more about this - I think it would be a good way to tie in the invasion of Proteron and/or Thurion: Zalek create the space, its then discovered and utilised by Proteron/Thurion by chance/experiment, and then manage to use it to access the Naev world and invade.

**Comment by onpon4** [comment]
Well no, the reason the Proteron are able to get to the jump gates is already decided: hypergates. See here: https://github.com/naev/naev/wiki/Spoilers#the-hypergate

The Za'lek could possibly utilize hypergates to access an alternate reality or something, but the Za'lek wouldn't directly cause the Proteron invasion, just be complicit in it.

🕵️

**Comment by agb32** [comment]
Yes, so I was thinking, the presence of hypergates could become possible by the Za'lek creating this alternative universe... 
So the Za'lek didn't cause it or even be complicit in it, but their invention enabled it to happen... (rather like a previous invention destroyed Sol...)

**Comment by onpon4** [comment]
I could see it as the source of the rediscovery of hypergates south of the Nebula, yes.

🕵️
